shale/src/lib.rs

```diff
@@ -206,11 +206,17 @@ impl<'a, T: Send + Sync> ObjRef<'a, T> {
 
         Ok(())
     }
+
+    pub fn into_ptr(self) -> DiskAddress {
+        let ptr = self.deref().as_ptr();
+        ptr
+    }
 }
 
 impl<'a, T: Send + Sync> Deref for ObjRef<'a, T> {
     type Target = Obj<T>;
     fn deref(&self) -> &Obj<T> {
+        // TODO: Something is seriously wrong here but I'm not quite sure about the best approach for the fix
```

Comment:
Not following. What's wrong? The unwrap()? Or the idea that there should be a deref on this at all?

Reply:
`unwrap` inside `deref` is horrible. The assumption is that something that implements `Deref` can simply be used as the `Deref` target. In other words, it's a wrapper type to  **strictly** adds functionality. In this case, `ObjRef` the "added" functionality is only valid if the type is initialized. In other words, it's "sometimes" added functionality. 

Anyway, I added to `into_ptr` method so that I don't have to manually drop the reference.